runelite-client/src/main/java/net/runelite/client/plugins/microbot/qualityoflife/scripts/WintertodtScript.java

```diff
@@ -0,0 +1,250 @@
+        return Rs2Player.getWorldLocation().getRegionID() == 6462;
+    }
+
+    public boolean run(QoLConfig config) {
+        WintertodtScript.config = config;
+        mainScheduledFuture = scheduledExecutorService.scheduleWithFixedDelay(() -> {
+            try {
+                if (!Microbot.isLoggedIn() || !super.run() || !isInWintertodtRegion()) {
+                    return;
+                }
+
+                Widget wintertodtHealthbar = Rs2Widget.getWidget(25952276);
+                isWintertodtAlive = Rs2Widget.hasWidget("Wintertodt's Energy");
+                if (wintertodtHealthbar != null && isWintertodtAlive) {
+                    String widgetText = wintertodtHealthbar.getText();
+                    wintertodtHp = Integer.parseInt(widgetText.split("\\D+")[1]);
+                } else {
+                    wintertodtHp = -1;
+                }
+                brokenBrazier = Rs2GameObject.getGameObjects(ObjectID.BRAZIER_29313).stream().findFirst().orElse(null);
+                unlitBrazier = Rs2GameObject.getGameObjects(ObjectID.BRAZIER_29312).stream().findFirst().orElse(null);
+
+                if (!config.interrupted())
+                    return;
+
+                NewMenuEntry actionToResume = config.wintertodtActions().getMenuEntry();
+                if (config.wintertodtActions().equals(WintertodtActions.FEED)) {
+                    GameObject fireBrazier = Rs2GameObject.getGameObjects(ObjectID.BURNING_BRAZIER_29314).stream().findFirst().orElse(null);
+                    if (fireBrazier != null && fireBrazier.getWorldLocation().distanceTo2D(Rs2Player.getWorldLocation()) < 5) {
+                        if (!Rs2Inventory.contains(ItemID.BRUMA_ROOT) && !Rs2Inventory.contains(ItemID.BRUMA_KINDLING)) {
+                            qolPlugin.updateLastWinthertodtAction(WintertodtActions.NONE);
+                            qolPlugin.updateWintertodtInterupted(false);
+                            Microbot.log("No resources found in inventory, cancelling action.");
+                            return;
+                        }
+                        Microbot.doInvoke(actionToResume, new java.awt.Rectangle(1, 1));
+                        qolPlugin.updateWintertodtInterupted(false);
+                    }
+                }
+                if (config.wintertodtActions().equals(WintertodtActions.FLETCH)) {
+                    if (!Rs2Inventory.contains(ItemID.BRUMA_ROOT)) {
+                        qolPlugin.updateLastWinthertodtAction(WintertodtActions.NONE);
+                        qolPlugin.updateWintertodtInterupted(false);
+                        return;
+                    }
+
+                    Microbot.doInvoke(actionToResume, new java.awt.Rectangle(1, 1));
+                    WintertodtActions.fletchBrumaRootsOnClicked();
+                    qolPlugin.updateWintertodtInterupted(false);
+                }
+
+            } catch (Exception ignore) {
+
```

Comment:
It might be good to log any exception that happeens in the main loop. Atleast the user will know something went wrong and will help in finding the issue